.github/workflows/ci.yml

```diff
@@ -1,39 +1,72 @@
-# This is a basic workflow to help you get started with MATLAB Actions
+#This workflow runs matrix build with both versions of setup MATLAB action hourly
 
-name: MATLAB Build
+name: MATLAB matrix build
 
-# Controls when the action will run. 
+# Controls when the action will run.
 on:
   push:
   pull_request:
   workflow_dispatch:
 
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  # This workflow contains a single job called "build-v1"
+  build-v1:
     # The type of runner that the job will run on
     runs-on: ubuntu-22.04
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v3
-      
+
       # Sets up MATLAB on the GitHub Actions runner
       - name: Setup MATLAB
         uses: matlab-actions/setup-matlab@v1
 
       # Runs a set of commands using the runners shell
-      - name: Run all tests
+      - name: Run tests and generate artifacts
         uses: matlab-actions/run-tests@v1
         with:
           source-folder: code
+          test-results-junit: test-results/results.xml
+          code-coverage-cobertura: code-coverage/coverage.xml
```

Comment:
Why are we producing result artifacts now? It doesn't look like they're being used anywhere. This may mess with Andy's runtime metrics.